1875_Mp3_Player_Clone.js

```diff
@@ -0,0 +1,316 @@
+
+createPlaylist() - Gets a name & ID. Creates a new, empty playlist with the given details. The ID is optional, and if omitted should be automatically generated. Returns the ID of the new playlist.
+
+playPlaylist() - Gets a playlist ID. Plays all songs in the specified playlist, in the order the appear in the playlist.
+
+editPlaylist() - Gets a playlist ID & a song ID. If the song ID exists in the playlist, removes it. If it was the only song in the playlist, also deletes the playlist. If the song ID does not exist in the playlist, adds it to the end of the playlist.
+
+playlistDuration() - Gets a playlist ID. Returns the total duration of the entire playlist with the given ID.
+
+searchByQuery() - Gets a query string. Returns a results object, which has:
+
+    -songs: an array of songs in which either title or album or artist contain the query string. The songs should be sorted by their titles.
+
+    -playlists: an array of playlists in which the name contains the query string. The playlists should be sorted by their names.
+
+searchByDuration() - Gets a duration in mm:ss format (for example 11:03). Returns the song, or playlist, with the closest duration to what was given.
+
+
+<------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------>
+`)
+
+const player = { //object of the 1875 mp3 player clone
+    songs: [
+      {
+        id: 1,
+        title: 'Boys and Girls',
+        album: 'CTV3',
+        artist: 'Jaden',
+        duration: 223
+      },
+      {
+        id: 2,
+        title: 'Dejanbem',
+        album: 'Haitian Boy Kodak',
+        artist: 'Kodak Black',
+        duration: 214
+      },
+      {
+        id: 7,
+        title: 'Versatile 3',
+        album: 'Before The Album',
+        artist: 'Kodak Black',
+        duration: 334
+      },
+      {
+        id: 3,
+        title: 'Gnarly',
+        album: 'Dying To Live',
+        artist: 'Kodak Black feat. Lil Pump',
+        duration: 214
+      },
+      {
+        id: 4,
+        title: 'Freaky Girl',
+        album: 'I AM YOU',
+        artist: 'YNW Melly',
+        duration: 215
+      },
+      {
+        id: 5,
+        title: 'Betrayal',
+        album: 'Trip At Knight',
+        artist: 'Trippie Redd feat. Drake',
+        duration: 151
+      }
+    ],
+    playlists: [
+      { id: 1, name: 'Hip-Hop', songs: [1, 3, 4] },
+      { id: 5, name: 'Rap', songs: [2, 7, 5] },
+    ],
+    playSong(song) {
+      console.log(`Playing ${song.title} from ${song.album} by ${song.artist} | ${sstomm(song.duration)}.`)
+    }
+  }
+  
+  function playSong(id) { //play a given song by its id
+    if(getSongIdIndex(id) === undefined) throw("song id not found in list");
+    player.playSong(player["songs"][getSongIdIndex(id)]); //uses the playSong method to play a specific song on the mp3!
```

Comment:
you are using getSongIdIndex(id) two times why not set it in variable

Reply:
true,